eahub/base/views.py

```diff
@@ -94,28 +94,77 @@ def groups(request):
     )
 
 
-def get_groups_data():
-    rows = Group.objects.all()
+def get_groups_data(user, by_distance=False):
+    if hasattr(user, "profile") and by_distance:
```

Comment:
I think it's a bit more performant/idiomatic to just assume that `user.profile` exists and then catch the `Profile.DoesNotExist` exception in case it doesn't. (Almost all users have profiles.)